tests/completions.sh

```diff
@@ -0,0 +1,48 @@
+source common.sh
+
+cd "$TEST_ROOT"
+
+mkdir -p dep && pushd dep
+cat <<EOF > flake.nix
+{
+    outputs = i: { };
+}
+EOF
+popd
+mkdir -p foo && pushd foo
+cat <<EOF > flake.nix
+{
+    inputs.a.url = "path:$(realpath ../dep)";
+
+    outputs = i: {
+        sampleOutput = 1;
+    };
+}
+EOF
+
+popd
+
+# Test the completion of a subcommand
+[[ $(printf "normal\nbuild\t\n") == $(NIX_GET_COMPLETIONS=1 nix buil) ]]
+[[ $(printf "normal\nmetadata\t\n") == $(NIX_GET_COMPLETIONS=2 nix flake metad) ]]
+
+# Filename completion
+[[ $(printf "filenames\n./foo\t\n") == $(NIX_GET_COMPLETIONS=2 nix build ./f) ]]
+[[ $(printf "filenames\n") == $(NIX_GET_COMPLETIONS=2 nix build ./nonexistent) ]]
+
+# Input override completion
+[[ $(printf "normal\na\t\n") == $(NIX_GET_COMPLETIONS=4 nix build ./foo --override-input '') ]]
+
+# Cli flag completion
+NIX_GET_COMPLETIONS=2 nix build --log-form | grep -- "--log-format"
```

Comment:
`grep -q` maybe?

Reply:
I like having the `grep` output for this kind of things since it's hidden by default but it gives a direct feedback as to whether the grep call found something or not when debugging (as it can be a bit tricky to understand which command exactly failed in the middle of a complex pipeline). Not that I have a very strong opinion about it though